sqlgen/db.go

```diff
@@ -361,6 +361,52 @@ func (db *DB) InsertRow(ctx context.Context, row interface{}) (sql.Result, error
 	return db.execWithTrace(ctx, query, "InsertRow")
 }
 
+// InsertRows inserts multiple rows into the database, chunksize rows at a time.
+// Most SQL db enforce a limit on max size of packet, which is why we need to break
+// the rows into chunks.
+//
+// rows should be an array of pointers to a struct, for example:
+//
+//   user1 := &User{Name: "foo"}
+//   user2 := &User{Name: "fan"}
+//   if err := db.InsertRows(ctx, []{interface{}}{user1, user2}, 100); err != nil {
+//
+func (db *DB) InsertRows(ctx context.Context, rows []interface{}, chunkSize int) (sql.Result, error) {
+	ctx, tx, err := db.WithTx(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	var result sql.Result
+	for j := 0; j < len(rows); j += chunkSize {
+		sliceLength := chunkSize
+		if len(rows) < j+chunkSize {
+			sliceLength = len(rows) - j
+		}
+		slice := rows[j:sliceLength]
+
+		query, err := db.Schema.MakeBatchInsertRow(slice)
+		if err != nil {
+			return nil, err
+		}
+
+		numColumns := len(query.Columns)
+		for i := 0; i < len(slice); i++ {
+			if err := db.checkColumnValuesAgainstLimits(ctx, query, query.Columns, query.Values[i*numColumns:(i+1)*numColumns], query.Table); err != nil {
+				return nil, err
+			}
+		}
+		result, err = db.execWithTrace(ctx, query, "InsertRows")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	tx.Commit()
```

Comment:
ah interesting. if a caller already has allocated a transaction in context, we shouldn't call Commit as it would commit any dirty write preceding InsertRows. not using transaction is also weird because I as a user would expect this function to write none or all since it's a single function call. 

maybe we check if we are called inside a transaction first and do things differently?

Reply:
good catch. changed to only start a transaction if one doesn't already exist